component/async/kafka/kafka.go

```diff
@@ -148,21 +147,6 @@ func (m *message) Raw() interface{} {
 	return m.msg
 }
 
-// DefaultSaramaConfig function creates a sarama config object with the default configuration set up.
-func DefaultSaramaConfig(name string) (*sarama.Config, error) {
```

Comment:
This will be the breaking change, right?

Reply:
The major breaking changes are that no Kafka consumer/producer can be created without providing Sarama configuration via `With*` options.

This one here is a *smaller* breaking change due to moving the default Sarama configs to the `v2` package; we could easily not make this breaking here, but if we don't then users will keep using this default Sarama configuration which would not force them to make a choice.

The non-breaking `DefaultSaramaConfig` would be like this:
```
func DefaultSaramaConfig(name string) (*sarama.Config, error) {
    return v2.DefaultConsumerSaramaConfig(name, false)
}
```

And would have read-committed disabled (as it is now).

What do you think? Prefer non-breaking or prefer force-choice? I opt for the latter since that is the reason we are making this PR.

Reply:
My concern is the following: If it is required to provide a sarama config it should not be added with an extension method but should be part of the constructor. If somebody does not provide it it will fail, which defeats the purpose.
My suggestion is to add the config to the constructor. We are already introducing breaking changes, we might as well do it properly. WDYT?

Reply:
Totally agree; I will move it to the constructor!